### ISSUE TITLE
Error serving cordova.js when I run phonegap serve

### ISSUE THREAD

**Issue by hoolymama** [phonegap/phonegap-cli#578]
I'm using phonegap 5.4.0 (also tried 5.4.1)

In order to get the deviceready event, I include cordova.js script in the head of index.html

when I run phonegap server I see the following errors in the terminal:

 TypeError: Path must be a string. Received undefined
    at assertPath (path.js:8:11)
    at Object.posix.join (path.js:477:5)
    at Object.handle (/usr/local/lib/node_modules/phonegap/node_modules/connect-phonegap/lib/middleware/cordova/cordova.js:25:33)
    at next (/usr/local/lib/node_modules/phonegap/node_modules/connect-phonegap/node_modules/connect/lib/proto.js:193:15)
    at resume (/usr/local/lib/node_modules/phonegap/node_modules/connect-phonegap/node_modules/connect/lib/middleware/static.js:65:7)
    at SendStream.error (/usr/local/lib/node_modules/phonegap/node_modules/connect-phonegap/node_modules/connect/lib/middleware/static.js:80:37)
    at emitOne (events.js:77:13)
    at SendStream.emit (events.js:169:7)
    at SendStream.error (/usr/local/lib/node_modules/phonegap/node_modules/connect-phonegap/node_modules/connect/node_modules/send/lib/send.js:147:51)
    at SendStream.onStatError (/usr/local/lib/node_modules/phonegap/node_modules/connect-phonegap/node_modules/connect/node_modules/send/lib/send.js:248:48)

So I looked at /usr/local/lib/node_modules/phonegap/node_modules/connect-phonegap/lib/middleware/cordova/cordova.js. It turns out req.session.device.version is undefined, and req.session.device.platform is set to the string "platform". So it never finds a cordova file.

Is this value supposed to be defined while using phonegap serve?

**Comment by mwbrooks** [comment]
Hey @hoolymama, that value should be defined.

What device are you using to access the app from `phonegap serve`? 

If possible, can you provide a code snippet - although it sounds like it's fine.

**Comment by hoolymama** [comment]
Hi @mwbrooks, I'm working on a Macbook Pro and I was connecting from iPad air2. 

I have since switched my workflow - I bought a $99 Apple developer license and running from Xcode, so I'm no longer using the developer app.

But I just tried again and, of course, the error has gone away. I don't think I changed very much - I removed the file and file transfer plugins.

Anyway - as I can't reproduce it, maybe its best to close.
